libnavigation-core/src/main/java/com/mapbox/navigation/core/internal/extensions/MapboxFlowableExtensions.kt

```diff
@@ -106,3 +110,25 @@ fun MapboxNavigation.flowNavigationSessionState(): Flow<NavigationSessionState>
     registerNavigationSessionStateObserver(observer)
     awaitClose { unregisterNavigationSessionStateObserver(observer) }
 }
+
+@OptIn(ExperimentalCoroutinesApi::class)
+fun MapboxNavigation.flowRouteAlternativeObserver():
+    Flow<Pair<RouteProgress, List<NavigationRoute>>> =
```

Comment:
Why don't you include RouterOrigin here?

Reply:
At the moment I did not see the need of including `RouterOrigin`, but this will definitely be revisited once we also support offline routing in Drop-In UI